V.2 Candidate

KarelWintersky opened the pull request:
* [+] More exceptions and error messages
* [+] Added ImageResizeInterface
* [*] Reorder and optimize code
* [*] GammaCorrection disabled by default (issue #162)
* [*] Reformat code
* [*] Tests: updated
* [*] Fixes: composer.json
* [*] Fixes: README.md

KarelWintersky commented:
Two months... 

bump?

peter279k commented:
Sorry. I fell of the stairs accidentally last Friday.

And I can't review your codes for now.

Once I recover, I will review your codes.

KarelWintersky commented:
Well, I'll fork this and publish as our corporate-internal package.

adityapatadia commented:
If you can wait till weekend, I can review and merge it.

peter279k commented on composer.json:
```diff
@@ -1,45 +1,43 @@
 {
-    "name": "gumlet/php-image-resize",
```
I don't know why the space tab has been changed.

KarelWintersky replied:
is it a problem?

peter279k replied:
I think it should keep previous style.

KarelWintersky replied:
have no time for this

KarelWintersky replied:
because code is reformatted

peter279k replied:
Never mind. This is about the small problem and it can be fixed on another PR.

KarelWintersky replied:
блядь, не можешь докопаться до сути - доебись до офорграфии.

peter279k commented on lib/ImageResize.php:
```diff
@@ -7,316 +7,333 @@
+     */
     public $source_type;
-
+    
```
Additional space tabs.

peter279k commented on lib/ImageResize.php:
```diff
@@ -7,316 +7,333 @@
         }
-        $resize = new self('data://application/octet-stream;base64,' . base64_encode($image_data));
-        return $resize;
+        return new self( 'data://application/octet-stream;base64,'.base64_encode( $image_data ) );
```
Please consider reverting the previous coding style.

KarelWintersky replied:
PHP compiler will optimize this code without any problems

KarelWintersky replied:
I can't understand you. You need in `$resize` local value?

peter279k commented on lib/ImageResize.php:
```diff
@@ -7,316 +7,333 @@
-        if (strstr(finfo_file($finfo, $filename), 'image') === false) {
-            if (version_compare(PHP_VERSION, '7.0.0', '<=') && strstr(file_get_contents($filename), 'WEBPVP8') !== false) {
-                $checkWebp = true;
+        if (!defined( 'IMAGETYPE_WEBP' )) {
```
Please consider reverting the previous coding style.

KarelWintersky replied:
code stylesheet required

peter279k commented on lib/ImageResizeInterface.php:
```diff
@@ -0,0 +1,218 @@
+    public function gamma($enable = true);
+    
+    public function imageFlip($image, $mode);
+}
```
Missing the EOF, end of file.

peter279k commented on lib/ImageResize.php:
```diff
@@ -712,78 +660,95 @@ protected function getCropPosition($expectedSize, $position = self::CROPCENTER)
         return $this;
     }
+    
+    
```
Additional new line.

KarelWintersky replied:
not an error

peter279k reviewed (changes requested):
Please consider following points:

- Reverting and be consistency for coding style.
- Checking the codes and remove useless new lines.

KarelWintersky commented:
Please, give me linter configuration.

KarelWintersky commented:
Please, add linter settings file

ties-s commented:
Any news on this? The docs claim chaining works but if I save multiple sizes this creates really dark images

KarelWintersky commented:
@ties-s , maintainers of gumlet/php-image-resize rejected any changes or pull requests. You can use my package: `ajur-media/php-image-resize`

peter279k commented:
Sorry to my delay response.

Most of stuffs are good, but it looks like a breaking change.

It defines the Interface and some coding styles are not good.

If I've available time, I will try to fix some stuffs by myself.

ties-s commented:
It’s about the gamma correction. While this may be a breaking change, it’s also a bug fix. The documentation claims chaining works fine, which it doesn’t right now. 
I think fixing this to align with docs and expected behavior should be a priority. You could always bump the version to indicate the breaking change.

peter279k commented:
> 
> 
> It’s about the gamma correction. While this may be a breaking change, it’s also a bug fix. The documentation claims chaining works fine, which it doesn’t right now.
> I think fixing this to align with docs and expected behavior should be a priority. You could always bump the version to indicate the breaking change.

I know that, but I need to review them carefully by myself.

Anyway, I will create another PR to work on that.

KarelWintersky commented:
> it’s also a bug fix. 

this bug fixed in my package ;-)

adityapatadia commented:
@KarelWintersky this was closed by you and we did not reject this. It would be best if you can create smaller PRs which are easy to review and we can merge them quickly.

I really appreciate your contribution but moving too many things in one go can sometimes take more time to review. Hope you can help by creating smaller PRs.

peter279k commented:
I agree with @adityapatadia because this PR is too huge to review. And most of code structured are changed a lot.

KarelWintersky commented:
Now I have no time for this, sorry. 

Our project is under of emergency migration from MySQL to Postgresql. Deadline avg 26 Dec 2021.

adityapatadia commented:
No problem. The gamma issue is fixed in latest version. The but of chaining not working will be resolved sometime soon.

KarelWintersky commented:
You delayed the process of reviewing my code for so long, and then made so many ridiculous and meaningless demands (by refusing to provide the linter configuration or a document about YOUR codestyle), that I lost all desire to help you fix errors in YOUR library.

You couldn't find time for more than two months to review changes that clearly improved the performance of your code and fixed many problems.

After that you made a lot of ridiculous claims about my work: “extra lines”, “extra spaces”, and so on and so forth.

Sorry, but I don't feel like investing in your library anymore. I corrected all the mistakes that you made during its development and released my own based on your code according to the license you specified.

You can use my code in your projects.
